source/iterator/iterator.go

```diff
@@ -110,29 +100,36 @@ func New(
 	}, nil
 }
 
-func setupSnapshotIterator(
-	ctx context.Context,
-	snowflake Repository,
-	table, key string,
-	columns []string,
-	element, offset, batchSize int,
-	isFirstStart bool,
-) (*SnapshotIterator, error) {
-	var (
-		index int
-	)
+func prepareCDC(ctx context.Context, snowflake *repository.Snowflake, table string) error {
+	// Check if table tracking table exist.
+	isTableExist, err := snowflake.IsTableExist(ctx, getTrackingTable(table))
+	if err != nil {
+		return fmt.Errorf("check if table exist: %w", err)
+	}
 
-	if !isFirstStart {
-		index = element + 1
+	if !isTableExist {
+		// Prepare tracking table for consume stream.
+		err = snowflake.CreateTrackingTable(ctx, getTrackingTable(table), table)
+		if err != nil {
+			return fmt.Errorf("create tracking table: %w", err)
+		}
 	}
 
-	data, err := snowflake.GetData(ctx, table, key, columns, offset, batchSize)
+	// Check if stream exist.
+	isStreamExist, err := snowflake.IsStreamExist(ctx, getTrackingTable(table))
 	if err != nil {
-		return nil, fmt.Errorf("get currentBatch: %w", err)
+		return fmt.Errorf("check if stream exist: %w", err)
 	}
 
-	return NewSnapshotIterator(snowflake, table,
-		columns, key, index, offset, batchSize, data), nil
+	if !isStreamExist {
+		// Prepare stream for cdc iterator.
+		err = snowflake.CreateStream(ctx, getStreamName(table), table)
+		if err != nil {
+			return fmt.Errorf("create stream: %w", err)
+		}
+	}
+
+	return nil
```

Comment:
Is it possible to write "create stream if not exists"?

Reply:
let me recheck docs

Reply:
let me recheck it

Reply:
yeah we can, I did it and removed not needed method